app/jobs/ingest_job.rb

```diff
@@ -4,11 +4,20 @@
 class IngestJob < ApplicationJob
   queue_as :default
 
-  def perform(druid:, background_job_result:)
+  # @param [String] druid
+  # @param [Array<Hash>] filesets the data for creating the structure
+  # @param [BackgroundJobResult] background_job_result
+  def perform(druid:, filesets:, background_job_result:)
     background_job_result.processing!
 
-    # TODO: generate contentMetadata.xml
-    # TODO: ship files
+    dir = StagingDirectory.new(druid: druid, staging_location: Settings.staging_location)
+
+    file_names = copy_files_to_staging(dir, filesets)
```

Comment:
Should this copy be verified? (That is, verify checksums).

Reply:
I don't think this is something we do when we do preassembly, so no.  If you feel strongly about it, open a ticket and we can discuss it with the PO.

Reply:
https://github.com/sul-dlss/google-books/issues/110